static/js/browser2/observation_chart.tsx

```diff
@@ -143,22 +176,11 @@ export class ObservationChart extends React.Component<
   };
 
   private handleDotClick = (dotData: DotDataPoint): void => {
-    // TODO (chejennifer): might need better way of getting dcid of observation node because this can be very slow
     const date = dotData.label;
-    let request = `/api/browser/observation-id?place=${this.props.placeDcid}&statVar=${this.props.statVarId}&date=${date}`;
-    if (this.props.sourceSeries.measurementMethod) {
-      request =
-        request +
-        `&measurementMethod=${this.props.sourceSeries.measurementMethod}`;
-    }
-    if (this.props.sourceSeries.observationPeriod) {
-      request =
-        request + `&obsPeriod=${this.props.sourceSeries.observationPeriod}`;
-    }
-    const obsDcidPromise = axios.get(request).then((resp) => resp.data);
-    obsDcidPromise.then((obsDcid) => {
+    const obsDcid = this.state.obsDcidMapping[date];
+    if (obsDcid) {
       const uri = URI_PREFIX + obsDcid;
       window.open(uri);
-    });
+    }
```

Comment:
should there be a response in the else case (error message perhaps?)

Reply:
I think in the original browser, the response was to take the user to the landing page of the browser tool. I do think an error message might be more useful, so I can try to do that

Reply:
added a simple error message: https://screenshot.googleplex.com/3RHNa8mwk4Zy6Fy